test/tests_for_integration/api_to_db_test.py

```diff
@@ -1191,21 +1317,214 @@ def test_run_job_batch(ee2_port, ws_controller, mongo_client):
                 "+KB_SOURCE_WS_OBJECTS": "",
             }
         )
+        _check_batch_htc_calls(
+            sub_init, schedd_init, sub, schedd, txn, expected_sub_1, expected_sub_2
+        )
+
+
+def test_run_job_batch_as_admin_with_job_reqs(ee2_port, ws_controller, mongo_client):
```

Comment:
I took a stab at factoring out some parts in common with the test above but it got pretty ugly and unreadable. Refactoring recommendations welcomed.